src/cryptsetup/cryptsetup.c

```diff
@@ -179,6 +180,13 @@ static int parse_one_option(const char *option) {
         } else if (streq(option, "tcrypt-system")) {
                 arg_type = CRYPT_TCRYPT;
                 arg_tcrypt_system = true;
+        } else if (streq(option, "tcrypt-veracrypt")) {
+#ifdef CRYPT_TCRYPT_VERA_MODES
+                arg_type = CRYPT_TCRYPT;
+                arg_tcrypt_veracrypt = true;
+#else
+                log_error("This version of cryptsetup does not support VeraCrypt; ignoring tcrypt-veracrypt.");
```

Comment:
Hmm.  I guess if I were a user, I would prefer bailing out completely with this message or similar.  The reasoning being that if I put `tcrypt-veracrypt` as an argument, I know that the regular Truecrypt mode won't work, and I'd rather get an error about "cryptsetup doesn't support that" rather than "can't find a valid header with that password."